pkg/api/cluster_client.go

```diff
@@ -79,7 +80,7 @@ func (c ClusterClient) ReadByName(ctx context.Context, name string) (*models.Clu
 	}
 
 	if len(clusters.Data) != 1 {
-		return &models.Cluster{}, Error404
+		return &models.Cluster{}, errors.New("resource Not Found")
```

Comment:
this should be a static error.
if an error is static, then we can make tests and compare concrete error types
if we generate the error at runtime like this, then it's more awkward to test, and we have to copy strings as in the test below